builder/openstack/image_query.go

```diff
@@ -0,0 +1,223 @@
+package openstack
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/images"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/packer/packer"
+)
+
+const (
+	descendingSort = "desc"
+	createdAtKey   = "created_at"
+)
+
+// Retrieve the specific ImageDateFilter using the exported const from images
+func getDateFilter(s string) (images.ImageDateFilter, error) {
+	filters := []images.ImageDateFilter{
+		images.FilterGT,
+		images.FilterGTE,
+		images.FilterLT,
+		images.FilterLTE,
+		images.FilterNEQ,
+		images.FilterEQ,
+	}
+
+	for _, filter := range filters {
+		if string(filter) == s {
+			return filter, nil
+		}
+	}
+
+	var badFilter images.ImageDateFilter
+	return badFilter, fmt.Errorf("No valid ImageDateFilter found for %s", s)
+}
+
+// Retrieve the specific ImageVisibility using the exported const from images
+func getImageVisibility(s string) (images.ImageVisibility, error) {
+	visibilities := [...]images.ImageVisibility{
+		images.ImageVisibilityPublic,
+		images.ImageVisibilityPrivate,
+		images.ImageVisibilityCommunity,
+		images.ImageVisibilityShared,
+	}
+
+	for _, visibility := range visibilities {
+		if string(visibility) == s {
+			return visibility, nil
+		}
+	}
+
+	var nilVisibility images.ImageVisibility
+	return nilVisibility, fmt.Errorf("No valid ImageVisilibility found for %s", s)
+}
+
+// Retrieve the specific ImageVisibility using the exported const from images
+func getImageStatus(s string) (images.ImageStatus, error) {
+	statuses := [...]images.ImageStatus{
+		images.ImageStatusActive,
```

Comment:
I think only `active` images are resonable. You can't/shouldn't use images in other states.

Reply:
I agree that makes sense. I was implementing it as a ListOpts parser and simply retrieving any potential value. I'll reduce the states to only "active"